src/query/encoding.ts

```diff
@@ -21,13 +21,34 @@ export interface EncodingQuery {
   bin?: boolean | BinQuery | ShortEnumSpec;
   scale?: boolean | ScaleQuery | ShortEnumSpec;
 
+  sort?: SortOrder | SortFieldQuery | ShortEnumSpec;
```

Comment:
The tricky part is here.  

We have to decide whether it's 

`SortOrder | Enumspec<SortOrder> | SortFieldQuery | ShortEnumSpec`

or 

`SortOrder | SortQuery | ShortEnumSpec` 

where SortQuery supports enumerating both SortOrder and SortField. 

The latter might be more correct, but also more complicated to handle.